File: Core/src/main/java/com/alphaz/core/authorization/model/permission/Operation.java
package com.alphaz.core.authorization.model.permission;


import com.alphaz.infrastructure.domain.constant.common.State;

/**
 * ProjectName: alphaz
 * PackageName: com.alphaz.core.pojo.dto
 * User: C0dEr
 * Date: 2017/7/7
 * Time: 下午5:10
 * Description:This is a class of com.alphaz.core.pojo.dto
 */
public class Operation {
    public Long id;
    public String operationname;
    public String label;
    public String icon;
    public State isenabled;

    public State getIsenabled() {
        return isenabled;
    }

    public void setIsenabled(State isenabled) {
        this.isenabled = isenabled;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getOperationname() {
        return operationname;
    }

    public void setOperationname(String operationname) {
        this.operationname = operationname;
    }

    public String getLabel() {
        return label;
    }

    public void setLabel(String label) {
        this.label = label;
    }

    public String getIcon() {
        return icon;
    }

    public void setIcon(String icon) {
        this.icon = icon;
    }
}

File: Core/src/main/kotlin/com/alphaz/core/authorization/permission/operation/Operation.kt
package com.alphaz.core.authorization.permission.operation

import com.alphaz.core.authorization.permission.Permission
import com.alphaz.infrastructure.domain.model.base.BaseDO
import javax.persistence.*

/**
 *@Author: c0der
 *@Date: 上午10:57 2018/3/16
 *@Description:
 */
@Entity
@Table(name = "alphaz_operation", catalog = "")
data class Operation(
        var operationName: String?,
        var label: String?,
        @OneToOne(targetEntity = Permission::class)
        @JoinColumn(name = "permission_id")
        var permission: Permission?,
        var icon: String?
) : BaseDO<Operation, Long>(), com.alphaz.infrastructure.domain.Entity {
    constructor() : this(null, null, null, null)

}